src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/SendingCsvWhenAgreedMediationSpecHandler.java

```diff
@@ -0,0 +1,118 @@
+import uk.gov.hmcts.reform.ccd.client.model.CallbackResponse;
+import uk.gov.hmcts.reform.civil.callback.Callback;
+import uk.gov.hmcts.reform.civil.callback.CallbackHandler;
+import uk.gov.hmcts.reform.civil.callback.CallbackParams;
+import uk.gov.hmcts.reform.civil.callback.CaseEvent;
+import uk.gov.hmcts.reform.civil.config.properties.robotics.RoboticsEmailConfiguration;
+import uk.gov.hmcts.reform.civil.model.CaseData;
+import uk.gov.hmcts.reform.civil.sendgrid.EmailAttachment;
+import uk.gov.hmcts.reform.civil.sendgrid.EmailData;
+import uk.gov.hmcts.reform.civil.sendgrid.SendGridClient;
+
+import java.math.BigDecimal;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static uk.gov.hmcts.reform.civil.callback.CallbackType.ABOUT_TO_SUBMIT;
+import static uk.gov.hmcts.reform.civil.callback.CaseEvent.SENDING_CSV_WHEN_AGREED_MEDIATION_SPEC;
+
+@Service
+@RequiredArgsConstructor
+public class SendingCsvWhenAgreedMediationSpecHandler extends CallbackHandler implements NotificationData {
+
+    private static final List<CaseEvent> EVENTS = List.of(SENDING_CSV_WHEN_AGREED_MEDIATION_SPEC);
+    public static final String TASK_ID = "SendingCsvWhenAgreedMediationSpec";
+    private final SendGridClient sendGridClient;
+    private final RoboticsEmailConfiguration roboticsEmailConfiguration;
+
+    @Override
+    protected Map<String, Callback> callbacks() {
+        return Map.of(
+            callbackKey(ABOUT_TO_SUBMIT), this::sendCVSMediation
+        );
+    }
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    @Override
+    public String camundaActivityId(CallbackParams callbackParams) {
+        return TASK_ID;
+    }
+
+    private CallbackResponse sendCVSMediation(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        Optional<EmailData> emailData = prepareEmail(caseData);
+        emailData.ifPresent(data -> sendGridClient.sendEmail(roboticsEmailConfiguration.getSender(), data));
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .build();
+    }
+
+    private Optional<EmailData> prepareEmail(CaseData data) {
+        String toEmail = "smallclaimsmediation@justice.gov.uk";
+        String subject = "OCMC Mediation Data";
+        String filename = "ocmc_mediation_data.csv";
+        String content = generateAttachmentContent(data);
+        InputStreamSource inputSource = new ByteArrayResource(content.getBytes(StandardCharsets.UTF_8));
+
+
+        return Optional.of(EmailData.builder()
+                               .to(toEmail)
+                               .subject(subject)
+                               .attachments(List.of(new EmailAttachment(inputSource, "text/csv", filename)))
+                               .build());
+    }
+
+    private String generateAttachmentContent(CaseData data) {
+
```

Comment:
this business logic should not be here. Perhaps create a service that would generate this and call it from here